fix: editor-view-dom type-check TS2339 (getAttribute on Node)

## Error
`packages/editor-view-dom type-check: src/dom-sync/dom-change-classifier.ts(1013,37): error TS2339: Property 'getAttribute' does not exist on type 'Node'.`

## Cause
`MutationRecord.target` is typed as `Node`; `getAttribute` exists only on `Element`.

## Fix
Narrow with `mutation.target instanceof Element` before calling `getAttribute`.

--- a/packages/editor-view-dom/src/dom-sync/dom-change-classifier.ts
+++ b/packages/editor-view-dom/src/dom-sync/dom-change-classifier.ts
@@ -1009,8 +1009,8 @@ function detectSpecialCase(
       .find((value): value is DomChangeCase => value !== null);
     if (directCase) return directCase;
 
-    if (mutation.attributeName) {
-      const attr = mutation.target?.getAttribute?.(mutation.attributeName);
+    if (mutation.attributeName && mutation.target instanceof Element) {
+      const attr = mutation.target.getAttribute(mutation.attributeName);
       if (attr && attr.toLowerCase().includes('autocorrect')) {
         return 'C4_AUTO_CORRECT';
       }

